frontend/src/components/StaticTokenKubeconfigConfiguration.vue

```diff
@@ -0,0 +1,74 @@
+<!--
+SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+<template>
+  <action-button-dialog
+    :shoot-item="shootItem"
+    @dialog-opened="onConfigurationDialogOpened"
+    ref="actionDialog"
+    width="450"
+    confirm-required
+    caption="Configure Admin Kubeconfig">
+    <template v-slot:actionComponent>
+      <static-token-kubeconfig-switch
+        v-model="enableStaticTokenKubeconfig"
+      ></static-token-kubeconfig-switch>
+    </template>
+  </action-button-dialog>
+</template>
+
+<script>
+import ActionButtonDialog from '@/components/dialogs/ActionButtonDialog'
+import StaticTokenKubeconfigSwitch from '@/components/StaticTokenKubeconfigSwitch'
+import { updateShootEnableStaticTokenKubeconfig } from '@/utils/api'
+import { errorDetailsFromError } from '@/utils/error'
+import shootItem from '@/mixins/shootItem'
+
+export default {
+  name: 'StaticTokenKubeconfigConfiguration',
+  components: {
+    ActionButtonDialog,
+    StaticTokenKubeconfigSwitch
+  },
+  mixins: [
+    shootItem
+  ],
+  data () {
+    return {
+      enableStaticTokenKubeconfig: undefined
+    }
+  },
+  methods: {
+    async onConfigurationDialogOpened () {
+      await this.reset()
+      const confirmed = await this.$refs.actionDialog.waitForDialogClosed()
+      if (confirmed) {
+        await this.updateConfiguration()
+      }
+    },
+    async updateConfiguration () {
+      try {
+        await updateShootEnableStaticTokenKubeconfig({
+          namespace: this.shootNamespace,
```

Comment:
I am not sure what it means from a performance and memory point of view to include a large mixin. The only thing we use is the `name` and `namespace` from the shootItem?

Reply:
IDK either, I also already thought about splitting the mixin up into separate parts as it grew a lot over time. In this case we also use `shootEnableStaticTokenKubeconfig`